### PR TITLE
Add audit task to ci

### LINKED ISSUE #18: Add auditing to CI
I think it would be interesting to add package auditing to the continuous integration to avoid (at least to filter `high` vulnerabilities) to keep our packages clean-ish.

### PR DESCRIPTION
**Pull request by xRuiAlves**
Closes #18

### CODE REVIEW

**Comment by xRuiAlves** [comment]
I have no idea what's failing in this build and this travis-ci is confusing af. Input @miguelpduarte ?

**Comment by xRuiAlves** [comment]
Also, how come the ci build takes 35 minutes to fail? Please let's change this to `circle-ci`, it's at least 35 time faster (since builds take at most 1 minute from my experience). Plus, it's better documented :x

**Comment by miguelpduarte** [comment]
> I have no idea what's failing in this build and this travis-ci is confusing af. Input @miguelpduarte ?

A snippet from one of the build logs:

```
$ git clone --depth=50 https://github.com/NIAEFEUP/nijobs-be.git NIAEFEUP/nijobs-be
Cloning into 'NIAEFEUP/nijobs-be'...
fatal: unable to access 'https://github.com/NIAEFEUP/nijobs-be.git/': Failed to connect to github.com port 443: Connection timed out

The command "eval git clone --depth=50 https://github.com/NIAEFEUP/nijobs-be.git NIAEFEUP/nijobs-be " failed. Retrying, 2 of 3.
```

I think GitHub was just down 😂 😂 (So the delay to fail was much bigger than expected, no way that the small amount of code in `develop` currently would run for 35 minutes for free hahaha)
Restarting the build manually after some OAuth(?) Let's see if it fixes that

File: .travis.yml [hunk 1]
```diff
@@ -16,7 +16,8 @@ env:
     # Any separate test environment variables go here, one per line each creating a different test environment
 
 script:
+  - npm audit --audit-level=high
```
Comment (miguelpduarte) [inline .travis.yml]: Not sure if this works due to the way we're using docker for some things, but let's see if the tests pass.

**Comment by imnotteixeira** [comment]
Managed to re-run some failing builds, but the missing one is actually failing because of one of the tests is failing. Track here: https://travis-ci.org/NIAEFEUP/nijobs-be/builds/568557192?utm_source=github_status

**Comment by GuilhermeJSilva** [comment]
The error comes from an after hook that erases all accounts in the test database, not from the actual tests. Running this tests locally resulted in 0 errors, in both the develop branch and this branch, however the hook was also badly designed, this hook actually erases the accounts after each test, which is not necessary, I think changing it to make the action before and after that specific collection of tests is more advisable. Furthermore, looking at the error it seems that it resulted from a timeout in the connection to the database

**Comment by miguelpduarte** [comment]
Now it seems that the tests are passing... Weird... :upside_down_face: This was probably due to some misconfiguration of docker stuff or of TravisCI (1 of the 4 checks is still failing, for some reason - shouldn't they all be the same? Travis is weird), I'm guessing.

Well, I'd say that this is just more motivation to tackle #19 and #21 :sweat_smile:

**Comment by miguelpduarte** [comment]
@xRuiAlves can you merge the changes from `develop` to here to see if the checks all pass?

**Comment by xRuiAlves** [comment]
Well, build was failing due to `stdout` "overflood" when running audit. Since we're only looking for the return status of the command, ignoring the text output did the trick

**Comment by imnotteixeira** [comment]
The acorn vulnerability has since been solved. @xRuiAlves Can you remove the `&> /dev/null` so that the output is shown?

**Review by imnotteixeira — APPROVED** [review]

**Pull request merged**